docs/guide/topics/permissions.rst

```diff
@@ -0,0 +1,219 @@
+
+.. code-block:: python3
+
+    # We can now create the overwrites mapping necessary for this, you can do it in one dictionary instantiation, or spead it across multiple lines.
+
+    # I have added type annotations here so that the reader can understand what is happeining, and so type-checkers do too:
+    overwrites: Dict[Union[discord.Role, discord.Member], discord.PermissionOverwrite] = {
+        guild.default_role: discord.PermissionOverwrite(view_channel=False, send_messages=False, read_messages=False, read_message_history=False),
+        moderator_role: discord.PermissionOverwrite(manage_messages=True),
+        member: discord.PermissionOverwrite(view_channel=True, send_messages=True, read_messages=True, read_message_history=True)
+    }
+
+    # or alternatively:
+    overwrites: Dict[Union[discord.Role, discord.Member], discord.PermissionOverwrite] = {}
+    overwrites[guild.default_role] = discord.PermissionOverwrite(view_channel=False, send_messages=False, read_messages=False, read_message_history=False)
+    overwrites[moderator_role] = discord.PermissionOverwrite(manage_messages=True)
+    overwrites[member] = discord.PermissionOverwrite(view_channel=True, send_messages=True, read_messages=True, read_message_history=True)
+
+    # We can now pass these to channel creation:
+
+    await guild.create_text_channel(name="Ticket", overwrites=overwrites)
+
+The channel created whose permissions look something like this:
+
+.. image:: /images/guide/topics/permissions/ticket_channel.png
+
+Using ``PermissionOverwrite`` objects for finer control over channels
+----------------------------------------------------------------------
+
+:class:`~PermissionOverwrite` makes granting finer access to channels easier to manipulate.
+Let's create a scenario where you want to "lockdown" a channel, denying the ``@everyone`` role the permission to send messages.
+
+.. code-block:: python3
+
+    # I have found that the easiest way to do this, is to get the current overwrites for the Role or Member we're editing:
+    overwrite = channel.overwrites_for(guild.default_role)
+    # We now have the PermissionOverwrite instance for the `@everyone` role, in the channel.
+
+    # All we do now, is update it. There are two ways to do this:
+    overwrite.send_messages = False
+    # or, if you have bulk permissions to update:
+    overwrite.update(send_messages=False)
+
+    # Since we have a singular PermissionOverwrite, we want to use the `set_permissions` method of the channel:
+    await channel.set_permissions(guild.default_role, overwrite=overwrite)
+
+Now, that channel has explicitly denied the ``send_messages`` permission to the defalt role of the guild.
+
+In an alternate scenario, we want to add multiple overwrites to a channel at the same time, to deny ``add_reactions`` permissions to roles and members:
+
+.. code-block:: python3
+
+    # Let's take a working copy of the overwrites in that channel
+    overwrites = channel.overwrites.copy()
```

Comment:
Is it necessary to do this? I've always seen copy as a user-facing thing for duplicating their own instances, so if it's necessary to do it here for some lib specific reason, we should probably explain why.